### PR TITLE
Fix wrongly exposed variables in `torch/__init__.py`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127710
* #127709
* #127708
* #127707
* #127706
* #127705
* #127704
* #127703
* __->__ #127795

image

This PR removes temporary variables in `torch/__init__.py`.

cc @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: torch/_dynamo/trace_rules.py [hunk 1]
```diff
@@ -2182,6 +2182,7 @@
         "torch.xlogy",
         "torch.zero_",
         "torch.zeros",
+        "torch.zeros_like",
```
Comment (XuehaiPan) [inline torch/_dynamo/trace_rules.py:2185]: `torch.zeros_like` was exposed as `torch.obj`, so it has no trace rule here.

**Review by albanD — APPROVED** [review]
Ho that is embarrassing,
We should be able to remove a few entries from https://github.com/pytorch/pytorch/blob/main/test/allowlist_for_publicAPI.json#L1329 now that this is fixed! It can be a follow up if needed.

**Comment by Skylion007** [comment]
python doc failures look real

**Pull request closed without merging**